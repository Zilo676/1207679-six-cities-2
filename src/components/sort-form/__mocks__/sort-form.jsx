import React from "react";

const SortForm = () => {
  return <></>;
};

export {SortForm};
export default SortForm;
