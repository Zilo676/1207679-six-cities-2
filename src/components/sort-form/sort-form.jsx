import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import { getSortType } from '../../reducer/hotels/selectors';

const SortForm = (props) => {
  const { cssClass, onClick, isActive, onForm, currentSortType } = props;
  const openedClass = cssClass + (isActive ? `__options--opened` : `__options--closed`);

  return (
    <form onClick={() => onForm()} className={`${cssClass}__sorting`} action="#" method="get">
      <span className={`${cssClass}__sorting-caption`}>Sort by </span>
      <span className={`${cssClass}__sorting-type`} tabIndex="0">
        {currentSortType}
        <svg className={`${cssClass}__sorting-arrow`} width="7" height="4">
          <use xlinkHref="#icon-arrow-select"></use>
        </svg>
      </span>
      <ul onClick={(evt) => onClick(evt)} className={`${cssClass}__options ${cssClass}__options--custom ${openedClass}`}>
        <li className={`${cssClass}__option`} tabIndex="0" >Popular</li>
        <li className={`${cssClass}__option`} tabIndex="0">Price: low to high</li>
        <li className={`${cssClass}__option`} tabIndex="0">Price: high to low</li>
        <li className={`${cssClass}__option`} tabIndex="0">Top rated first</li>
      </ul>

    </form>
  );
}

SortForm.propTypes = {
  cssClass: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  onForm: PropTypes.func,
  isActive: PropTypes.bool.isRequired,
  currentSortType: PropTypes.string.isRequired,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  currentSortType: getSortType(state),
});

export { SortForm };

export default connect(mapStateToProps, null)(SortForm);

