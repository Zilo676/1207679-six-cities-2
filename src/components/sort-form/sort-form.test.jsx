import React from 'react';
import renderer from 'react-test-renderer';

import {SortForm} from './sort-form.jsx';

it(`Correctly rendered`, () => {
  const mockFunc = jest.fn();
  const tree = renderer.create(<SortForm
    cssClass={`TEST`}
    onClick={mockFunc}
    onForm={mockFunc}
    isActive={true}
    currentSortType={`Sort Type`}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
