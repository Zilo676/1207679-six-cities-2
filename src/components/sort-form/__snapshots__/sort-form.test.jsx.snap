// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly rendered 1`] = `
<form
  action="#"
  className="TEST__sorting"
  method="get"
  onClick={[Function]}
>
  <span
    className="TEST__sorting-caption"
  >
    Sort by 
  </span>
  <span
    className="TEST__sorting-type"
    tabIndex="0"
  >
    Sort Type
    <svg
      className="TEST__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="TEST__options TEST__options--custom TEST__options--opened"
    onClick={[Function]}
  >
    <li
      className="TEST__option"
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="TEST__option"
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="TEST__option"
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="TEST__option"
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
